extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/DatasourceServiceBindingConfigSourceFactory.java

```diff
@@ -79,23 +79,38 @@ protected String formatUrl(String urlFormat, String type, String host, String da
         return String.format(urlFormat, type, host, portPart, database);
     }
 
+    private boolean configureUrlPropertyUsingKey(Map<String, String> properties, String key) {
+        String value = serviceBinding.getProperties().get(key);
+        if (value != null) {
+            if (value.startsWith(prefix)) {
+                properties.put(urlPropertyName, value);
+                return true;
+            } else {
+                log.warnf("The value '%s' from the property '%s' does not start with '%s'. It will be ignored.",
+                        value, key, prefix);
+            }
+        }
+
+        return false;
```

Comment:
```suggestion
       if (value == null) {
            return false;
        }
        if (value.startsWith(prefix)) {
            properties.put(urlPropertyName, value);
            return true;
        }
        log.warnf("The value '%s' from the property '%s' does not start with '%s'. It will be ignored.",
        value, key, prefix);
```

Reply:
suggestion to simplify the logic

Reply:
Updated